Fix compilation issues related to libraries and flags
--- a/configure.ac
+++ b/configure.ac
@@ -62,7 +62,7 @@
 	    echo "$1 LD $2"  >>$ff_where_lib_conf
 	    ff_WHERE_INC=`AS_DIRNAME(["$3"])`
 	    case "$ff_WHERE_INC" in 
-		/*)e cho "$1 INCLUDE -I$ff_WHERE_INC"  >>$ff_where_lib_conf ;;
+		/*)echo "$1 INCLUDE -I$ff_WHERE_INC"  >>$ff_where_lib_conf ;;
 	    esac
 	fi
 	AC_MSG_RESULT($ff_WHERE)
@@ -1744,9 +1744,9 @@
 	[[Superlu4],[-lsuperlu4],[/usr/include/superlu4/superlu_enum_consts.h],[]],
         [[blacs],[ -lblacsCinit$ff_with_mpi	-lblacsF77init$ff_with_mpi	-lblacs$ff_with_mpi],[]],
         [[scalapack],[-lscalapack$ff_with_mpi],[]],
-        [[scotch],[-lscotch -lscotcherr],[]],
+        [[scotch],[-lscotch -lscotcherr],[/usr/include/scotch/scotch.h][]],
         [[ptscotch],[-lptscotch -lptscotcherr],[]],
-	[[metis],[-lscotch -lmetis],[/usr/include/metis/metis.h],[]],
+	[[metis],[-lscotch -lmetis],[/usr/include/metis.h],[]],
 	[[metis],[-lscotch -lmetis],[metis.h],[]],
 	[[parmetis],[-lptscotch -lparmetis],[],[]],
 	[[freeyams],[-lfreeyams],[freeyamslib.h],[]],
