Index: freefem++-2.20/configure.ac
===================================================================
--- freefem++-2.20.orig/configure.ac	2007-10-19 13:33:30.466147130 +0200
+++ freefem++-2.20/configure.ac	2007-10-19 13:37:14.478912890 +0200
@@ -1052,7 +1052,7 @@
 AC_CHECK_PROG(ff_dvips,dvips,yes,no)
 # to translate the figure 
 AC_CHECK_PROG(ff_pdf2ps,pdf2ps,yes,no)
-AC_CHECK_PROG(ff_epstopdf,epstopdf,yes,no)
+AC_CHECK_PROGS(EPSTOPDF,[epstopdf pstopdf],[false])
 AC_CHECK_PROG(ff_convert,convert,yes,no)
 if test "$ff_latex" = yes -a "$ff_makeindex" = yes -a "$ff_dvips" = yes -a $ff_pdf2ps = yes -a $ff_convert = yes;
 then
@@ -1072,7 +1072,7 @@
 if test "$enable_pdf" = yes 
 then
    AC_CHECK_PROG(ff_pdflatex,pdflatex,yes,no)
-   if test "$ff_pdflatex" = yes -a $ff_epstopdf = yes -a $ff_convert = yes;
+   if test "$ff_pdflatex" = yes -a $EPSTOPDF != false -a $ff_convert = yes;
    then
 	AC_SUBST(DOCPDF,"freefem++doc.pdf")
    fi
Index: freefem++-2.20/DOC/Makefile.am
===================================================================
--- freefem++-2.20.orig/DOC/Makefile.am	2007-10-19 13:33:30.466147130 +0200
+++ freefem++-2.20/DOC/Makefile.am	2007-10-19 13:35:01.471333220 +0200
@@ -31,9 +31,9 @@
 	-rm -r cpfigs
 
 cpfigs/%.pdf:plots/%.eps
-	epstopdf $^ --outfile=$@
+	$(EPSTOPDF) $^ --outfile=$@
 cpfigs/%.pdf:figures/%.eps
-	epstopdf $^ --outfile=$@
+	$(EPSTOPDF) $^ --outfile=$@
 cpfigs/%.eps:plots/%.jpg
 	convert $^  $@
 cpfigs/%.eps:plots/%.pdf
@@ -115,4 +115,4 @@
 
 
 # do not delete the copy figure to long to created 
-.PRECIOUS: $(CPFIGS_PDF) $(CPFIGS_EPS)
\ No newline at end of file
+.PRECIOUS: $(CPFIGS_PDF) $(CPFIGS_EPS)
