Index: freefem++-3.20/examples++-mpi/hips_FreeFem.cpp
===================================================================
--- freefem++-3.20.orig/examples++-mpi/hips_FreeFem.cpp	2013-02-08 15:07:48.000000000 +0100
+++ freefem++-3.20/examples++-mpi/hips_FreeFem.cpp	2013-02-08 15:20:39.000000000 +0100
@@ -45,8 +45,8 @@
 #endif
 
 extern "C" {
-#include "hips.h"  
-#include "metis.h"
+#include "hips/hips.h"  
+#include "metis/metis.h"
 }
 #include <stdio.h>
 #include <stdlib.h>
Index: freefem++-3.20/examples++-mpi/complex_pastix_FreeFem.cpp
===================================================================
--- freefem++-3.20.orig/examples++-mpi/complex_pastix_FreeFem.cpp	2012-10-18 13:07:39.000000000 +0200
+++ freefem++-3.20/examples++-mpi/complex_pastix_FreeFem.cpp	2013-02-08 15:23:31.000000000 +0100
@@ -55,7 +55,7 @@
 #include <sys/time.h>
 
 
-#include "pastix_long_complex.h"
+#include "pastix/pastix_long_complex.h"
 
 #undef memFree_null
 #define memFree_null(x) {if (x ==NULL) {fprintf(stdout,"%s:%d freeing NULL\n",__FILE__,__LINE__);} free(x); x=NULL;}
Index: freefem++-3.20/examples++-mpi/real_pastix_FreeFem.cpp
===================================================================
--- freefem++-3.20.orig/examples++-mpi/real_pastix_FreeFem.cpp	2012-10-18 13:07:39.000000000 +0200
+++ freefem++-3.20/examples++-mpi/real_pastix_FreeFem.cpp	2013-02-08 15:22:53.000000000 +0100
@@ -76,7 +76,7 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include "pastix.h"
+#include "pastix/pastix.h"
 //#include "cscd_utils.h"
 #ifdef __cplusplus
 }
