Index: freefem++-3.9-2/configure.ac
===================================================================
--- freefem++-3.9-2.orig/configure.ac	2010-09-07 12:47:24.000000000 +0200
+++ freefem++-3.9-2/configure.ac	2010-09-07 12:48:29.000000000 +0200
@@ -1734,10 +1734,13 @@
 ff_save_libs=$LIBS
 LIBS="$LIBS $2"
 AC_LINK_IFELSE([AC_LANG_PROGRAM([m4_if($3,,,[#include <$3>])],[])], [ff_WHERE=yes],[ff_WHERE=no]) 
-if test "$ff_where" = "yes" ; then
-     echo "$1 LD '$2'"  >$ff_where_lib_conf
+if test "$ff_WHERE" = "yes" ; then
+     echo "$1 LD '$2'"  >>$ff_where_lib_conf
 if test -n "$3" ; then
-       echo "$1 INCLUDE  /usr/include/AS_DIRNAME[$3]  >$ff_where_lib_conf
+       if test "$incdir" != "."; then
+          echo "$1 INCLUDE " -I/usr/include/$incdir
+          echo "$1 INCLUDE " -I/usr/include/$incdir  >>$ff_where_lib_conf
+       fi
 fi 
 fi
 AC_MSG_RESULT($ff_WHERE)
@@ -1749,10 +1752,11 @@
 m4_map([AC_FF_WHERELIB],[ 
 [[mumps],[-lpord -lmumps_common -ldmumps -lzmumps -lpord],[dmumps_c.h]],
 [[fftw3],[-lfftw3],[fftw3.h]],
-[[superlu],[-lsuperlu],[superlu_defs.h]],
-[[blacs],[ -lblacsCinit$ff_with_mpi	-lblacsF77init$ff_with_mpi	-lblacs$ff_with_mpi],[]],
-[[scalapack],[-lscalapack],[]],
-[[scotch],[-lscotch -lscotcherr],[]],	
+[[superlu],[-lsuperlu],[superlu/slu_ddefs.h]],
+[[blacs],[ -lblacsCinit$ff_with_mpi	-lblacsF77init$ff_with_mpi	-lblacs$ff_with_mpi]],
+[[scalapack],[-lscalapack$ff_with_mpi]],
+[[scotch],[-lscotch -lscotcherr]],
+[[ptscotch],[-lptscotch -lptscotcherr]],
 [[metis],[-lmetis],[metis/metis.h]],
 [[parmetis],[-lparmetis],[parmetis/parmetis.h]]
 ]
