Index: freefem++-3.9-2/examples++-load/ff-pkg-download.in
===================================================================
--- freefem++-3.9-2.orig/examples++-load/ff-pkg-download.in	2010-09-07 12:56:53.000000000 +0200
+++ freefem++-3.9-2/examples++-load/ff-pkg-download.in	2010-09-07 12:57:11.000000000 +0200
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 ff=$0;
 bb=`basename $0`
 DIR=`dirname $ff`
@@ -53,4 +55,4 @@
     done
 else
     echo error not directory $DIR/lib and $DIR/include  $DIRP 
-fi
\ No newline at end of file
+fi
