Index: freefem++-2.20/configure.ac
===================================================================
--- freefem++-2.20.orig/configure.ac	2007-10-19 13:19:05.416850780 +0200
+++ freefem++-2.20/configure.ac	2007-10-19 13:19:29.918247035 +0200
@@ -892,7 +892,7 @@
 
 	# Trying default locations
 
-	AC_CHECK_HEADERS(umfpack.h umfpack/umfpack.h ufsparse/umfpack.h,ff_umfpack_header=yes)
+	AC_CHECK_HEADERS(umfpack.h umfpack/umfpack.h ufsparse/umfpack.h suitesparse/umfpack.h,ff_umfpack_header=yes)
 
 	# Somes systems like FreeBSD hide umfpack.h in a directory
 	# called UMFPACK (all capitals). This breaks the standard
@@ -909,7 +909,7 @@
 	if test "$ff_amd_ok" = no;
 	then
 		AC_CHECK_LIB(amd,amd_info,
-			ff_umfpack_lib="-lamd"
+			ff_umfpack_lib="-lamd -llapack -lblas"
 			ff_amd_ok=yes)
 	fi
 
Index: freefem++-2.20/src/femlib/MatriceCreuse.hpp
===================================================================
--- freefem++-2.20.orig/src/femlib/MatriceCreuse.hpp	2007-10-19 13:19:05.416850780 +0200
+++ freefem++-2.20/src/femlib/MatriceCreuse.hpp	2007-10-19 13:19:29.918247035 +0200
@@ -18,10 +18,14 @@
 #ifdef HAVE_UFSPARSE_UMFPACK_H
 #include <ufsparse/umfpack.h>
 #else
+#ifdef HAVE_SUITESPARSE_UMFPACK_H
+#include <suitesparse/umfpack.h>
+#else
 
   // Defaults to a local version of the UMFPACK headers
 #include "../../download/include/umfpack.h"
 
+#endif // HAVE_SUITESPARSE_UMFPACK_H
 #endif // HAVE_UFSPARSE_UMFPACK_H
 #endif // HAVE_BIG_UMFPACK_UMFPACK_H
 #endif // HAVE_UMFPACK_UMFPACK_H
